src/stretchy.js

```diff
@@ -0,0 +1,179 @@
+    xmapsto : "\u21a6",
+    xrightharpoondown : "\u21c1",
+    xleftharpoondown : "\u21bd",
+    xrightleftharpoons : "\u21cc",
+    xleftrightharpoons : "\u21cb",
+    xtwoheadleftarrow : "\u219e",
+    xtwoheadrightarrow : "\u21a0",
+    xLongequal : "=",
+    xtofrom : "\u21c4",
+};
+
+const mathMLnode = function(label) {
+    const node = new mathMLTree.MathNode(
+        "mo", [new mathMLTree.TextNode(stretchyCodePoint[label.substr(1)])]);
+    node.setAttribute("stretchy", "true");
+    return node;
+};
+
+const katexImagesData = {
+                 // height, depth, fileName
+    overleftarrow : [0.334, 0, "leftarrow"],
+    underleftarrow : [0.334, 0, "leftarrow"],
+    xleftarrow : [0.261, 0.261, "xleftarrow"],
+    overrightarrow : [0.334, 0, "rightarrow"],
+    underrightarrow : [0.334, 0, "rightarrow"],
+    xrightarrow : [0.261, 0.261, "xrightarrow"],
+    overbrace : [0.548, 0, "overbrace"],
+    underbrace : [0.548, 0, "underbrace"],
+    overbracket : [0.3, 0, "overbracket"],
+    underbracket : [0.3, 0, "underbracket"],
+    overleftrightarrow : [0.334, 0, "leftrightarrow"],
+    underleftrightarrow : [0.334, 0, "leftrightarrow"],
+    xleftrightarrow : [0.261, 0.261, "xleftrightarrow"],
+    Overrightarrow : [0.56, 0, "doublerightarrow"],
+    xLeftarrow : [0.28, 0.28, "doubleleftarrow"],
+    xRightarrow : [0.28, 0.28, "doublerightarrow"],
+    xLeftrightarrow : [0.28, 0.28, "doubleleftrightarrow"],
+    overleftharpoon : [0.334, 0, "leftharpoon"],
+    overrightharpoon : [0.334, 0, "rightharpoon"],
+    xleftharpoonup : [0.261, 0.261, "xleftharpoon"],
+    xrightharpoonup : [0.261, 0.261, "xrightharpoon"],
+    xhookleftarrow : [0.261, 0.261, "hookleftarrow"],
+    xhookrightarrow : [0.261, 0.261, "hookrightarrow"],
+    overlinesegment : [0.334, 0, "linesegment"],
+    underlinesegment : [0.334, 0, "linesegment"],
+    xmapsto : [0.261, 0.261, "mapsto"],
+    xrightharpoondown : [0.261, 0.261, "xrightharpoondown"],
+    xleftharpoondown : [0.261, 0.261, "xleftharpoondown"],
+    xrightleftharpoons : [0.358, 0.358, "rightleftharpoons"],
+    xleftrightharpoons : [0.358, 0.358, "leftrightharpoons"],
+    overgroup : [0.262, 0, "overgroup"],
+    undergroup : [0.262, 0, "undergroup"],
+    xtwoheadleftarrow : [0.167, 0.167, "twoheadleftarrow"],
+    xtwoheadrightarrow : [0.167, 0.167, "twoheadrightarrow"],
+    xLongequal : [0.167, 0.167, "longequal"],
+    xtofrom : [0.264, 0.264, "tofrom"],
+};
+
+const svgSpan = function(group, options) {
+    // Create a span with class(es) that refer to the background-image
+    // and/or the mask-image.
+    const label = group.value.label.substr(1);
+    let height = 0;
+    let depth = 0;
+    const classArray = ["stretchy"];
+    let fileName = "";
+
+    if (utils.contains(["widehat", "widetilde", "undertilde"], label)) {
+        // There are four SVG images available for each function.
+        // Choose a taller image when there are more characters.
```

Comment:
Per *The TeXbook*, page 443, rule 12:

TeX has a series of glyphs available for `\widetilde` and `\widehat`. It calculates the length of the subject matter and uses the shortest glyph that is longer than the subject matter. When the subject matter is longer than any of the glyphs, TeX puts in the longest one it has. So there can be a mis-match in length, like this: ![](http://quicklatex.com/cache3/78/ql_43a59253e4e5380e961760eff45d6978_l3.png).

I believe the MnSymbols package has more and longer glyphs to pick from, but it also eventually runs out of glyphs.

My basic scheme doesn't know when to stop. It will match the length of any subject matter.  And these images start to look flat when the aspect ratio gets too small. So for long strings, I use a taller image.